### DISCUSSION TITLE
Feature Request: Support for Downloading Non-JSON Response Bodies (e.g., CSV, ZPL)

### DISCUSSION THREAD

**Discussion by 0pilatos0** [scalar/scalar#1709] [Ideas]
I've encountered a limitation regarding the handling of non-JSON response bodies like CSV, ZPL, etc. Currently, the interface doesn’t support direct insights into or interactions with these types of files.

**Feature Request:**
I propose adding functionality that allows users to download response bodies of various file types directly from the documentation interface. This feature would not only extend the utility of [Project Name] but also enhance user experience by accommodating a wider range of API responses.

**Proposed Solution:**
Implement a download button or link for response types that cannot be directly rendered in the browser. This approach would handle file types such as CSV, ZPL, XML, etc., where the content can be more usefully accessed via a local file rather than an in-browser display.

**Comment by hanspagel** [comment]
Yesss! In the works :))

**Reply by hanspagel** [reply]
Btw, a download button/action would be great for all content types, eg json

**Comment by afuetterer** [comment]
I am not sure, if my question is the same, but it is definitively related, so I put it here.
I would like to see a preview of the response data in the UI, if it comes as `text/xml`. Is that possible?
